include/cpp2util.h

```diff
@@ -2281,8 +2281,15 @@ struct range
         : first{ f }
         , last{ l }
     {
-        if (include_last) {
-            ++last;
+        if (include_last) { 
+            if constexpr (std::is_arithmetic_v<T>) {
```

Comment:
Perhaps `is_integral` (or `std::integral`)

Reply:
Just because floats shouldn't come into these anyway, since they're not reliably equality comparable... and maybe there should also be a check for that with a static error if they try to make a range of floats?

Reply:
Makes sense to me. Done.